pkg/cloudprovider/provider/cherryservers/provider.go

```diff
@@ -0,0 +1,333 @@
+		}
+	}
+
+	return &cherryServer{server: server}, nil
+}
+
+type cherryServer struct {
+	server cherrygo.Server
+}
+
+func (s *cherryServer) Name() string {
+	return s.server.Hostname
+}
+
+func (s *cherryServer) ID() string {
+	return strconv.Itoa(s.server.ID)
+}
+
+func (s *cherryServer) Addresses() []string {
+	var addresses []string
+	for _, ip := range s.server.IPAddresses {
+		addresses = append(addresses, ip.Address)
+	}
+	return addresses
+}
+
+func (s *cherryServer) Status() instance.Status {
+	if s.server.State != "active" {
+		return instance.StatusCreating
+	}
+	return instance.StatusRunning
+}
+
+func (p *provider) Get(machine *v1alpha1.Machine, _ *cloudprovidertypes.ProviderData) (instance.Instance, error) {
+	c, _, err := p.getConfig(machine.Spec.ProviderSpec)
+	if err != nil {
+		return nil, cloudprovidererrors.TerminalError{
+			Reason:  common.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, due to %v", err),
+		}
+	}
+
+	client := getClient(c.Token)
+
+	servers, _, err := client.Servers.List(c.ProjectID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, server := range servers {
+		if server.Hostname == machine.Name {
+			srv, _, err := client.Server.List(strconv.Itoa(server.ID))
+			if err != nil {
+				return nil, err
+			}
+			return &cherryServer{server: srv}, nil
+		}
+	}
+
+	return nil, cloudprovidererrors.ErrInstanceNotFound
+}
+
+func (p *provider) MigrateUID(machine *v1alpha1.Machine, new types.UID) error {
+	return nil
+}
+
+func (p *provider) GetCloudConfig(spec v1alpha1.MachineSpec) (config string, name string, err error) {
+	return "", "", nil
+}
+
+func (p *provider) SetMetricsForMachines(machines v1alpha1.MachineList) error {
+	return nil
+}
+
+func (p *provider) MachineMetricsLabels(machine *v1alpha1.Machine) (map[string]string, error) {
+	labels := make(map[string]string)
+
+	c, _, err := p.getConfig(machine.Spec.ProviderSpec)
+	if err == nil {
+		labels["plan"] = c.Plan
+		labels["location"] = c.Location
+	}
+
+	return labels, err
+}
+
+func (p *provider) Cleanup(machine *v1alpha1.Machine, data *cloudprovidertypes.ProviderData) (bool, error) {
+	server, err := p.Get(machine, data)
+	if err != nil {
+		if err == cloudprovidererrors.ErrInstanceNotFound {
+			return true, nil
+		}
+		return false, err
+	}
+
+	c, _, err := p.getConfig(machine.Spec.ProviderSpec)
+	if err != nil {
+		return false, cloudprovidererrors.TerminalError{
+			Reason:  common.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, due to %v", err),
+		}
+	}
+
+	client := getClient(c.Token)
+
+	_, _, err = client.Server.Delete(
+		&cherrygo.DeleteServer{
+			ID: server.ID(),
+		})
+	if err != nil {
+		return false, cloudprovidererrors.TerminalError{
+			Reason:  common.DeleteMachineError,
+			Message: fmt.Sprintf("Could not delete machine, due to %v", err),
+		}
+	}
+
+	return false, nil
+}
+
+func NewKey() (privateKey []byte, publicKey string, err error) {
```

Comment:
Is this function used anywhere? I wasn't able to find any reference, but I might be mistaken.

Reply:
My bad, not used anymore after removing ssh plumbing.